src/AgentModel.cpp

```diff
@@ -155,206 +159,261 @@ void AgentModel::decisionProcessStop() {
         e.position = INFINITY;
         e.standingTime = INFINITY;
     }
-   
 
- //add new signals (added trafficlight extension)
-    unsigned int i = 0;
+    // add stop point because of destination point
+    if (_input.signals[agent_model::NOS-1].id == agent_model::NOS)
+    {
+        _state.decisions.stopping[agent_model::NOS-1].id = agent_model::NOS;
+        _state.decisions.stopping[agent_model::NOS-1].position = _input.vehicle.s + _input.signals[agent_model::NOS-1].ds;
+        _state.decisions.stopping[agent_model::NOS-1].standingTime = INFINITY;
+    }
+
+    // not yet decided about to stop or drive
+    bool stop = false;
+    bool drive = false;
+    bool found_signal = false;
+
+    // iterate over all signals and mark ds of the next relavant signal
+    double ds_rel_tls = INFINITY;
+    double ds_rel_sgn = INFINITY;
+    agent_model::Signal* rel;
+    agent_model::Signal* rel_tls;
+    agent_model::Signal* rel_sgn;
 
     for(auto &e : _input.signals) 
     {
-        //set destination stop
-        if(e.id == agent_model::NOS) 
+        if (e.id == agent_model::NOS) continue;
```

Comment:
Maybe we should make the "convention" about the destination stop more clear. 
Is it the last entry in the signals array? Or id==NOS? Or both?
In my opinion the ID is irrelevant and we just take the last entry. In our OSI adapter we just use NOS as an id, since our ids are counted from 1 to NOS-1

Reply:
That would mean this loop should omit the last entry

Reply:
Right now, it is both. Correct, the last entry would be sufficient, but however we have to set an ID. So maybe setting agent_model::NOS as ID is a good choice here.

Reply:
We have to make a decision and document it in Interface.h for the case someone wants to fill the SimDriver interface. 
If we just say the last entry is reserved for the destination then we don't need to care about any IDs, or am I wrong?

Reply:
--> Use horizon for destination. 

New field `double destination`